core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1796,6 +1853,15 @@ object Parser extends ParserInstances {
       override def parseMut(state: State): A = oneOf(ary, state)
     }
 
+    case class StringIn1(all: List[String]) extends Parser1[Unit] {
```

Comment:
maybe we should accept a SortedSet[String] here, which makes it structurally deterministic, and then we can skip having sorted, and just all.toList will be a sorted list we can pass to the RadixNode. What do you think?